pyproject.toml

```diff
@@ -63,6 +63,7 @@ hypothesis = "^6.54.5"
 ipython = "^8.5.0"
 sphinx = ">=6,<8"
 babelgladeextractor = "^0.7"
+ruff = "^0.3.5"
```

Comment:
Normally we run ruff through `pre-commit run --all-files`. Is there an advantage to adding it as a dependency?

Reply:
hi @danyeaw, thanks for the appreciation! Indeed, there is no need to add this; it was some time after the commit in which I added ruff to the toml that I realized how to use pre-commit. I'll remove it.